examples/02_matrix_vector_product_basic.cpp

```diff
@@ -31,11 +31,11 @@ int main(int argc, char* argv[]) {
     mdspan<double, extents<std::size_t, dynamic_extent>> y(y_vec.data(),N);
     for(int i=0; i<A.extent(0); i++)
       for(int j=0; j<A.extent(1); j++)
-        A(i,j) = 100.0*i+j;
+        A[i,j] = 100.0*i+j;
```

Comment:
This will only compile if the C++23 feature "multidimensional subscript operator" (P2128R6) is available.  As a result, would you consider one of the following changes?

1. Protect the example so it only compiles if the feature test macro `__cpp_multidimensional_subscript` is defined, OR
2. Add `#define MDSPAN_USE_PAREN_OPERATOR 1` to the example before including any `mdspan` headers (see top of https://godbolt.org/z/Yrr8oe9sE for an example of the relevant macros), and use parentheses instead of brackets (e.g., `A(i,j)`)